crates/paralegal-flow/src/ana/graph_converter.rs

```diff
@@ -352,30 +369,14 @@ impl<'a, 'tcx, C: Extend<DefId>> GraphConverter<'tcx, 'a, C> {
     fn handle_node_types(&mut self, old_node: Node, weight: &DepNode<'tcx>) {
         let i = self.new_node_for(old_node);
 
-        let is_controller_argument =
-            matches!(self.try_as_root(weight.at), Some(l) if l.location == RichLocation::Start);
-
-        if self
-            .dep_graph
-            .graph
-            .edges_directed(old_node, Direction::Incoming)
-            .any(|e| e.weight().target_use.is_return() && e.weight().source_use.is_argument())
-        {
-            assert!(
-                weight.place.projection.is_empty(),
-                "{:?} at {} has projection",
-                weight.place,
-                weight.at
-            );
-        } else if !is_controller_argument {
-            return;
+        let place_ty = self.determine_place_type(weight.at, weight.place.as_ref());
+        let place_info = self.place_info(weight.at.leaf().function);
+        let deep = !place_info.children(weight.place).is_empty();
+        let mut node_types = self.type_is_marked(place_ty, deep).collect::<HashSet<_>>();
+        for (p, _) in weight.place.iter_projections() {
+            let place_ty = self.determine_place_type(weight.at, p);
+            node_types.extend(self.type_is_marked(place_ty, false));
```

Comment:
@willcrichton This hunk is the crux of the algorithm in that it iterates our ancestors to pick up markers and it determines whether we need to include markers from our children.